deploy.py

```diff
@@ -272,776 +286,471 @@ def notarizefile(filetonotarize, username, password):
+                          f'rpm package at \"{build_filename}\" is ' + Fore.GREEN + 'ready' + Fore.CYAN + \
+                          ' for distribution.'
+        print(f"{deployed_message}", flush=True)
 
-        buildFilename=f'bin/{buildArch}/Deploy/rpm/RPMS/{buildArch}/pingnoo-{rpmVersion}-{rpmRelease}.{buildArch}.rpm'
 
-        deployedMessage = deployedMessage + f'\r\n' + Style.BRIGHT + Fore.CYAN + f'rpm package at \"{buildFilename}\" is '+ Fore.GREEN+'ready' + Fore.CYAN + ' for distribution.'
+def _do_windows():
+    """ Windows version """
 
+    def win_sign_binary(signingtool, filetosign, _unused_cert, timeserver, pin=None):
+        # TODO: Move to new execute
+        # FIXME: cert parameter was unused?
+        if pin:
+            pin = "-pin " + pin
 
-    endTime = time.time()
+        return execute(f'{signingtool} {pin} sign /fd sha256 /t {timeserver} /a "{filetosign}"')
 
-    # done!
-
-    print(f"{deployedMessage}", flush=True)
-
-    print(Style.BRIGHT + f'\r\nTotal time taken to perform deployment was ' +
-          timedelta(endTime - startTime) + '.', flush=True)
-
-    exit(0)
-
-if platform.system() == "Darwin":
-    print(Style.BRIGHT+'Deployment process started at '+str(datetime.datetime.now())+'\r\n', flush=True)
-
-    startTime = time.time()
+    with msg_printer('Checking for curl...'):
+        if args.curlbin and os.path.isfile(args.curlbin):
+            curl = args.curlbin
+        else:
+            curl = which('curl.exe')
+        if not curl:
+            raise MsgPrinterException('curl could not be found. (see --curlbin).')
 
     # check for qt installation
-
-    startmessage('Checking qtdir...')
-
-    if args.qtdir:
-        if os.path.isfile(args.qtdir+'/bin/qmake'):
-            qtdir = args.qtdir
+    with msg_printer('Checking qtdir...'):
+        if args.qtdir and os.path.isfile(f'{args.qtdir}\\bin\\windeployqt.exe'):
+            windeployqt = f'{args.qtdir}\\bin\\windeployqt.exe'
         else:
-            qtdir = None
-    else:
-        qmake = which('qmake')
+            windeployqt = which('windeployqt')
 
-        if qmake:
-            qtdir = parent(os.path.normpath(os.path.dirname(qmake)))
-        else:
-            qtdir = None
+        if not windeployqt:
+            raise MsgPrinterException('qt could not be found. (see --qtdir).')
 
-    if not qtdir:
-        endmessage(False, 'qt directory could not be found. (see --qtdir).')
-        exit(1)
-    else:
-        if not os.path.isdir(qtdir):
-            endmessage(False, 'qt directory could not be found. (see --qtdir).')
-            exit(1)
+    tempdir = os.path.normpath(tempfile.mkdtemp())
+    signtool = args.signtool
 
-    endmessage(True)
+    if args.cert:
+        if not os.path.exists(signtool):
+            with msg_printer('Downloading SmartCardTools...'):
+                execute(f'cd \"{tempdir}\" && \"{curl}\"'
+                        f' -LJO https://www.mgtek.com/files/smartcardtools.zip',
```

Comment:
I _think_ a lot of the f-strings are redundant; there is no field there.

Reply:
Yeah, I'll run through these and remove where it doesn't make sense.